### ISSUE TITLE
Compiling v0.3.1-alpha2

### ISSUE THREAD

**Issue by rzerres** [redox-os/orbtk#222]
**Compile v0.3.1-alpha2 on arch-linux**
Arch offers a stable compiler package (rustc v1.44.0)
I have cloned/pulled updates for Orbtk as usual.
When compiling  examples or the doc, compiler will throw unrecoverable errors.
May that is expected because of the api redesign?  Should i simply wait for updates?
**To Reproduce**
1. cd ~/orbtk
2. cargo run --example widgets
3. await compiling the dependencies

**Expected behavior**
Should create the given crates

**Error messages**
```sh
compiling orbtk-api v0.3.1-alpha2 (/data/development/redox-os/orbtk/crates/api)
error[E0107]: wrong number of type arguments: expected 3, found 2
  --> crates/api/src/application/window.rs:21:16
   |
21 |     pub world: World<Tree, StringComponentStore>,
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments
```

``` sh
Compiling orbtk-api v0.3.1-alpha2 (/data/development/redox-os/orbtk/crates/api)
error[E0107]: wrong number of type arguments: expected 3, found 2
  --> crates/api/src/application/window.rs:21:16
   |
21 |     pub world: World<Tree, StringComponentStore>,
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments

error[E0107]: wrong number of type arguments: expected 3, found 2
  --> crates/api/src/systems/cleanup_system.rs:17:6
   |
17 | impl System<Tree, StringComponentStore> for CleanupSystem {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments

error[E0107]: wrong number of type arguments: expected 3, found 2
   --> crates/api/src/systems/event_state_system.rs:323:6
    |
323 | impl System<Tree, StringComponentStore> for EventStateSystem {
    |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments

error[E0107]: wrong number of type arguments: expected 3, found 2
  --> crates/api/src/systems/init_system.rs:44:6
   |
44 | impl System<Tree, StringComponentStore> for InitSystem {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments

error[E0107]: wrong number of type arguments: expected 3, found 2
  --> crates/api/src/systems/layout_system.rs:13:6
   |
13 | impl System<Tree, StringComponentStore> for LayoutSystem {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments

error[E0107]: wrong number of type arguments: expected 3, found 2
  --> crates/api/src/systems/post_layout_state_system.rs:17:6
   |
17 | impl System<Tree, StringComponentStore> for PostLayoutStateSystem {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments

error[E0107]: wrong number of type arguments: expected 3, found 2
  --> crates/api/src/systems/render_system.rs:22:6
   |
22 | impl System<Tree, StringComponentStore> for RenderSystem {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0107`.
error: could not compile `orbtk-api`.
```

**Comment by FloVanGH** [comment]
It looks like it references an old version of DCES. Could you check if `cargo update` helps? Where did you loaded the example from?

**Comment by rzerres** [comment]
@FloVanGH I downloaded the code from the github repository.
Then i chrosschecked with the repros at gitlab. Since that i saw an updated branch (theme) i gave it a shot. And it does compile. So probably this issue will render obsolete, once you upload the stuff.
For now, i have a starting point to dig into the api changes coming with the theme integration. 

I thought the primary development is shifted to github. Did you go back to gitlab?

**Comment by FloVanGH** [comment]
Yes the complete development is moved to github, the gitlab repo is now only a mirror. It is planned to stay with OrbTk on github.

**Comment by rzerres** [comment]
Since the issue is fixed in the upcoming  v0.3.1-alpha3 i will close this issue.
